2024-07-11 12:27:54.655676: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:27:54.655867: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:28:03,190 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:28:03,301 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:28:03,301 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:28:03,301 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:28:04,465 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:28:04,465 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:04,482 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4166666666666667}
INFO flwr 2024-07-11 12:28:04,482 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:28:04,722 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:04,750 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:28:04,752 | fedavg.py:242 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:04,760 | server.py:131 | fit progress: (1, 0.36504071577322433, {'accuracy': 0.8466666666666667}, 0.2773462000000002)
DEBUG flwr 2024-07-11 12:28:04,760 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:04,782 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:28:04,782 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:28:04,782 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:28:04,806 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:04,817 | server.py:131 | fit progress: (2, 0.3623439486516099, {'accuracy': 0.85}, 0.3342457000000003)
DEBUG flwr 2024-07-11 12:28:04,818 | server.py:178 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:04,842 | server.py:192 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:04,842 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:04,869 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:04,879 | server.py:131 | fit progress: (3, 0.3633755947885282, {'accuracy': 0.8533333333333334}, 0.39712639999999944)
DEBUG flwr 2024-07-11 12:28:04,880 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:04,898 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:28:04,899 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:28:04,927 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:04,938 | server.py:131 | fit progress: (4, 0.36337538869027114, {'accuracy': 0.8533333333333334}, 0.4550748999999996)
DEBUG flwr 2024-07-11 12:28:04,938 | server.py:178 | evaluate_round 4: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:04,961 | server.py:192 | evaluate_round 4 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:28:04,961 | server.py:227 | fit_round 5: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:28:05,001 | server.py:241 | fit_round 5 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:05,012 | server.py:131 | fit progress: (5, 0.3695539233970718, {'accuracy': 0.8433333333333334}, 0.5294743000000004)
DEBUG flwr 2024-07-11 12:28:05,013 | server.py:178 | evaluate_round 5: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,040 | server.py:192 | evaluate_round 5 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:05,040 | server.py:227 | fit_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,076 | server.py:241 | fit_round 6 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:05,088 | server.py:131 | fit progress: (6, 0.37081488571188936, {'accuracy': 0.84}, 0.6061268999999996)
DEBUG flwr 2024-07-11 12:28:05,089 | server.py:178 | evaluate_round 6: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,116 | server.py:192 | evaluate_round 6 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:05,116 | server.py:227 | fit_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,159 | server.py:241 | fit_round 7 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:05,172 | server.py:131 | fit progress: (7, 0.37081476671665214, {'accuracy': 0.84}, 0.6897757000000002)
DEBUG flwr 2024-07-11 12:28:05,173 | server.py:178 | evaluate_round 7: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,201 | server.py:192 | evaluate_round 7 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:05,201 | server.py:227 | fit_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,241 | server.py:241 | fit_round 8 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:05,254 | server.py:131 | fit progress: (8, 0.37081451406264665, {'accuracy': 0.84}, 0.7714411000000005)
DEBUG flwr 2024-07-11 12:28:05,255 | server.py:178 | evaluate_round 8: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,287 | server.py:192 | evaluate_round 8 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:05,287 | server.py:227 | fit_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,326 | server.py:241 | fit_round 9 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:05,339 | server.py:131 | fit progress: (9, 0.3708145065317643, {'accuracy': 0.84}, 0.8568894999999994)
DEBUG flwr 2024-07-11 12:28:05,339 | server.py:178 | evaluate_round 9: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,374 | server.py:192 | evaluate_round 9 received 6 results and 0 failures
DEBUG flwr 2024-07-11 12:28:05,374 | server.py:227 | fit_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,422 | server.py:241 | fit_round 10 received 6 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:28:05,434 | server.py:131 | fit progress: (10, 0.3708145061127231, {'accuracy': 0.84}, 0.9512546999999998)
DEBUG flwr 2024-07-11 12:28:05,434 | server.py:178 | evaluate_round 10: strategy sampled 6 clients (out of 6)
DEBUG flwr 2024-07-11 12:28:05,468 | server.py:192 | evaluate_round 10 received 6 results and 0 failures
INFO flwr 2024-07-11 12:28:05,469 | server.py:153 | FL finished in 0.9857606000000008
INFO flwr 2024-07-11 12:28:05,469 | app.py:225 | app_fit: losses_distributed [(1, 0.3777739691237609), (2, 0.3824386606613795), (3, 0.3819651593764623), (4, 0.39641535693201524), (5, 0.3970169437783105), (6, 0.3969033428600856), (7, 0.39690414667129514), (8, 0.39690415007727486), (9, 0.39690414709704264), (10, 0.39690414709704264)]
INFO flwr 2024-07-11 12:28:05,469 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:28:05,469 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:28:05,470 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.36504071577322433), (2, 0.3623439486516099), (3, 0.3633755947885282), (4, 0.36337538869027114), (5, 0.3695539233970718), (6, 0.37081488571188936), (7, 0.37081476671665214), (8, 0.37081451406264665), (9, 0.3708145065317643), (10, 0.3708145061127231)]
INFO flwr 2024-07-11 12:28:05,470 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.4166666666666667), (1, 0.8466666666666667), (2, 0.85), (3, 0.8533333333333334), (4, 0.8533333333333334), (5, 0.8433333333333334), (6, 0.84), (7, 0.84), (8, 0.84), (9, 0.84), (10, 0.84)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
>>>>>>>>>>>>> evaluating on server-side
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.11462246,  1.07990021,  0.80047382,  0.30411446,  0.24589894,
         0.06092977, -0.04998476,  0.12862145]]), array([0.08181501])]
[array([[-2.11462246,  1.07990021,  0.80047382,  0.30411446,  0.24589894,
         0.06092977, -0.04998476,  0.12862145]]), array([0.08181501])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.12408326,  1.05523381,  0.87941735,  0.33541347,  0.24280829,
         0.05573391, -0.10142005,  0.14074813]]), array([0.08716122])]
[array([[-2.12408326,  1.05523381,  0.87941735,  0.33541347,  0.24280829,
         0.05573391, -0.10142005,  0.14074813]]), array([0.08716122])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.09246309,  1.07461277,  0.86798753,  0.37545916,  0.22965406,
         0.08360546, -0.09742108,  0.08263789]]), array([0.03243256])]
[array([[-2.09246309,  1.07461277,  0.86798753,  0.37545916,  0.22965406,
         0.08360546, -0.09742108,  0.08263789]]), array([0.03243256])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.09242147,  1.07467233,  0.86790054,  0.37552538,  0.22956878,
         0.08360334, -0.09751245,  0.08267763]]), array([0.03246732])]
[array([[-2.09242147,  1.07467233,  0.86790054,  0.37552538,  0.22956878,
         0.08360334, -0.09751245,  0.08267763]]), array([0.03246732])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.0276402 ,  1.07531888,  0.77805186,  0.38632336,  0.20469615,
         0.13160617, -0.05441161, -0.02976846]]), array([0.07705674])]
[array([[-2.0276402 ,  1.07531888,  0.77805186,  0.38632336,  0.20469615,
         0.13160617, -0.05441161, -0.02976846]]), array([0.07705674])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.02696496,  1.06973403,  0.75369604,  0.36787513,  0.17396404,
         0.13695769, -0.06528507, -0.02577731]]), array([0.16096599])]
[array([[-2.02696496,  1.06973403,  0.75369604,  0.36787513,  0.17396404,
         0.13695769, -0.06528507, -0.02577731]]), array([0.16096599])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.0264247 ,  1.0695264 ,  0.75363729,  0.36765938,  0.17388944,
         0.13691805, -0.06538532, -0.02561587]]), array([0.16080859])]
[array([[-2.0264247 ,  1.0695264 ,  0.75363729,  0.36765938,  0.17388944,
         0.13691805, -0.06538532, -0.02561587]]), array([0.16080859])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.0264239 ,  1.06952779,  0.75363674,  0.36766596,  0.17389009,
         0.13691619, -0.06538894, -0.02561351]]), array([0.16080244])]
[array([[-2.0264239 ,  1.06952779,  0.75363674,  0.36766596,  0.17389009,
         0.13691619, -0.06538894, -0.02561351]]), array([0.16080244])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.02642385,  1.06952778,  0.75363677,  0.36766605,  0.17389016,
         0.13691621, -0.06538899, -0.02561343]]), array([0.16080212])]
[array([[-2.02642385,  1.06952778,  0.75363677,  0.36766605,  0.17389016,
         0.13691621, -0.06538899, -0.02561343]]), array([0.16080212])]
>>>>>>>>>>>>> evaluating on server-side
[array([[-2.02642385,  1.06952778,  0.75363678,  0.36766605,  0.17389017,
         0.13691621, -0.06538899, -0.02561342]]), array([0.1608021])]
[array([[-2.02642385,  1.06952778,  0.75363678,  0.36766605,  0.17389017,
         0.13691621, -0.06538899, -0.02561342]]), array([0.1608021])]
>>>>>>>>>>>>>run time:
0.453125s
